source/MaterialXTest/MaterialXCore/CoreUtil.cpp

```diff
@@ -36,6 +37,28 @@ TEST_CASE("String utilities", "[coreutil]")
     REQUIRE(!mx::stringStartsWith("testName", "Name"));
     REQUIRE(mx::stringEndsWith("testName", "Name"));
     REQUIRE(!mx::stringEndsWith("testName", "test"));
+
+    std::string inputScalar1 = "  00.1000  ";
```

Comment:
This covers all the variations I could come up with.

Reply:
Looks like a pretty complete list to me. 

Though when I read it - I find myself wondering if MaterialX supports parsing scientific notation - but thats perhaps out of scope of this particular PR to address.

Reply:
I've put in a check for exponent ('e') and skip trailing 0 removal.